settings.gradle

```diff
@@ -1,3 +1,23 @@
+pluginManagement {
+    resolutionStrategy {
+        eachPlugin {
+            // Work around https://github.com/gradle/gradle/issues/1697.
+            if (requested.version == null) {
+                def versionPropertyName = (requested.id.id == 'org.jetbrains.kotlin.jvm') ?
+                        "kotlinVersion" : "${requested.id.name}Version"
```

Comment:
Question to the reviewers: Would you prefer a "PluginVersion" suffix here, to separate plugin versions from "normal" dependency versions in `gradle.properties`?

Reply:
You mean like `detektPluginVersion` instead of `detektVersion`? Sounds good, but I think we should also separate plugin versions from normal dependencies in `gradle.properties`, and then it's not that important to have the `plugin` suffix.

Reply:
Consider also prefixing in addition e.g.: 'gradleKotlinPluginVersion' 
This yields a plugins being grouped when properties are sorted lexicographically

Reply:
Please see the current version of the PR, I've simply created a separate "block" in `gradle.properties`, and each block is sorted alphabetically.